src/Host/Client/Impl/Definitions/IRSessionProvider.cs

```diff
@@ -5,19 +5,28 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.R.Host.Client.Host;
 
 namespace Microsoft.R.Host.Client {
     public interface IRSessionProvider : IDisposable {
-        IRSession GetOrCreate(Guid guid);
+        IRSession GetOrCreate(Guid guid, IRHostBrokerConnector brokerConnector);
```

Comment:
Yes, if there is more than one REPL. `IRHostBrokerConnector` is assigned to the `RInteractiveWorkflow`, while `IRSessionProvider` is a global export

---
In reply to: [73030385](https://github.com/Microsoft/RTVS/pull/2133#discussion_r73030385) [](ancestors = 73030385)